### PR TITLE
Scaffolder Gitlab Merge Requests - added assignee as an optional field

### PR DESCRIPTION
**Pull request by ljiljanaholden**
## Hey, I just made a Pull Request!

Added assignee as an optional input to Gitlab Merge Request action. 
Gitlab expects this field to be a Gitlab-specific user id, so we first perform a Gitlab User lookup based on the passed username and then use the returned id to identify MR assignee. 
If a user with a given username does not exist, the error message is written in the log, and the MR creation process will proceed. 
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlabMergeRequest.ts [hunk 1]
```diff
@@ -146,6 +152,21 @@ export const createPublishGitlabMergeRequestAction = (options: {
...
+      const assignee = ctx.input.assignee;
+
+      let assigneeId = undefined;
+
+      if (assignee !== undefined) {
+        try {
+          const assigneeUser = await api.Users.username(assignee);
+          assigneeId = assigneeUser[0].id;
+        } catch (e) {
+          throw new InputError(
```
Comment (benjdlambert) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlabMergeRequest.ts:164]: Do we actually want to fail the flow here? Or rather just log that we couldn't assign it to a user but still create the MR?

**Review by benjdlambert — COMMENTED** [review]
Question about the behaviour here. Maybe we just log if it fails instead? Feels like a waste to have to run the job again because it wasn't assigned to a person properly?

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlabMergeRequest.ts (hunk 1 above)
Comment (ljiljanaholden) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlabMergeRequest.ts:164]: Good point, I'll make the change now

**Review by benjdlambert — APPROVED** [review]
Ok this is looking nice now! :pray:

**Pull request merged**